app/controllers/users_controller.rb

```diff
@@ -0,0 +1,29 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:danger] = t ".not_found"
+    redirect_to root_path
+  end
```

Comment:
```suggestion
  def show; end
```
Tách logic bên trong hàm show ra 1 hàm load_user, rồi call trong before_action